go/keyset/manager.go

```diff
@@ -64,12 +90,115 @@ func (km *Manager) Rotate(kt *tinkpb.KeyTemplate) error {
+func (km *Manager) SetPrimary(keyID uint32) error {
+	if km.ks == nil {
+		return fmt.Errorf("keyset_manager: cannot set primary, no keyset")
+	}
+	for i := 0; i < len(km.ks.Key); i++ {
+		if km.ks.Key[i].KeyId == keyID {
+			if km.ks.Key[i].Status == tinkpb.KeyStatusType_ENABLED {
+				km.ks.PrimaryKeyId = keyID
+				return nil
+			}
+			return fmt.Errorf("keyset_manager: cannot set key as primary because it's not enabled")
+		}
+	}
+	return fmt.Errorf("keyset_manager: key not found %d", keyID)
+}
+
+// Enable will enable the key with given keyID
+// returns an error if the key is not found or is not enabled or disabled already
+func (km *Manager) Enable(keyID uint32) error {
+	if km.ks == nil {
+		return fmt.Errorf("keyset_manager: cannot enable key, no keyset")
+	}
+	for i := 0; i < len(km.ks.Key); i++ {
+		if km.ks.Key[i].KeyId == keyID {
+			if km.ks.Key[i].Status == tinkpb.KeyStatusType_ENABLED || km.ks.Key[i].Status == tinkpb.KeyStatusType_DISABLED {
+				km.ks.Key[i].Status = tinkpb.KeyStatusType_ENABLED
+				return nil
+			}
+			return fmt.Errorf("keyset_manager: cannot enable key with id %d", keyID)
+		}
+	}
+	return fmt.Errorf("keyset_manager: key not found %d", keyID)
+}
+
+// Disable will disable the key with given keyID
+// returns an error if the key is not found or it is the primary key
+func (km *Manager) Disable(keyID uint32) error {
+	if km.ks == nil {
+		return fmt.Errorf("keyset_manager: cannot disable key, no keyset")
+	}
+	if km.ks.PrimaryKeyId == keyID {
+		return fmt.Errorf("keyset_manager: cannot disable the primary key")
+	}
+	for i := 0; i < len(km.ks.Key); i++ {
+		if km.ks.Key[i].KeyId == keyID {
+			if km.ks.Key[i].Status == tinkpb.KeyStatusType_ENABLED || km.ks.Key[i].Status == tinkpb.KeyStatusType_DISABLED {
+				km.ks.Key[i].Status = tinkpb.KeyStatusType_DISABLED
+				return nil
+			}
+			return fmt.Errorf("keyset_manager: cannot disable key with id %d", keyID)
+		}
+	}
+	return fmt.Errorf("keyset_manager: key not found %d", keyID)
+}
+
+// Delete will delete the key with given keyID, removing the key from the keyset entirely
+// returns an error if the key is not found or it is the primary key
+func (km *Manager) Delete(keyID uint32) error {
+	if km.ks == nil {
+		return fmt.Errorf("keyset_manager: cannot delete key, no keyset")
+	}
+	if km.ks.PrimaryKeyId == keyID {
+		return fmt.Errorf("keyset_manager: cannot delete the primary key")
+	}
+	// simpler to copy elements to new keys object
+	var keys []*tinkpb.Keyset_Key
+	var found bool
+	for _, key := range km.ks.Key {
+		if key.KeyId == keyID {
+			found = true
+			continue
+		}
+		keys = append(keys, key)
+	}
+	if !found {
+		return fmt.Errorf("keyset_manager: key not found %d", keyID)
+	}
+	km.ks.Key = keys
 	return nil
 }
 
+// Destroy will destroy will the key material associated with a given keyID
+// returns an error if the key is not found or it is the primary key
+func (km *Manager) Destroy(keyID uint32) error {
```

Comment:
This method should be removed because I don't think anybody needs it.

Do you?

Reply:
I can live without Delete (I think).

Reply:
Sorry, I agree would like to have either Delete or Destroy to cleanup old keys, I wasn't paying that close of attention. Isn't either of them available in the Java/C++ APIs?

Reply:
OK let's keep Delete and remove Destroy.